### PR TITLE
minor fix to consume review comments for 39135

### PR DESCRIPTION
**Pull request by irisdingbj**

### CODE REVIEW

**Comment by irisdingbj** [comment]
@costinm This is the follow-up PR to land your comments in #39135 , Please take a review, Thanks.

**Review by GregHanson — APPROVED** [review]

File: manifests/charts/istiod-remote/files/injection-template.yaml [hunk 1]
```diff
@@ -371,8 +371,8 @@ spec:
     volumeMounts:
     - name: workload-socket
       mountPath: /var/run/secrets/workload-spiffe-uds
-    - name: credential-socket
-      mountPath: /var/run/secrets/credential-uds
+    - name: gateway-cert-socket
```
Comment (howardjohn) [inline manifests/charts/istiod-remote/files/injection-template.yaml:374]: Why is it called "gateway"? This is a sidecar

Comment (irisdingbj) [reply manifests/charts/istiod-remote/files/injection-template.yaml:374]: make sense, will `external-cert-socket` be more approriate?

**Pull request closed without merging**